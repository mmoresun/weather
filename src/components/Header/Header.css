.header__main {
    width: 100vw;
    background-color: rgb(113, 141, 189);
    padding: 10px;
    margin: 0 0 20px 0;
    color: white;
    text-align: center;    
}