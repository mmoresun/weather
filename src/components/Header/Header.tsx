import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';

import './Header.css';

export default function Header() {
  return (

    <Box sx={{ flexGrow: 1, margin: 1, paddingBottom: 2 }}>
      <AppBar position="static" sx={{ background: 'rgb(113, 141, 189)' }}>
        <Toolbar sx={{}}>
          <IconButton
            size="large"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{ mr: 2, visibility: 'hidden' }}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            <a href="https://mmoresun.github.io/weather" style={{ textDecoration: 'none', color: 'white' }}>&#9925; Weather Anywhere</a>
          </Typography>
          <Button color="inherit" size='small' href='https://github.com/mmoresun/weather/tree/master' target='_blank'>GitHub</Button>
        </Toolbar>
      </AppBar>
    </Box>
  );
}