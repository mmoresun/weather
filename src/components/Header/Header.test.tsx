
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Header from './Header';

it('renders Header correctly (snapshot)', () => {
    const tree = renderer
        .create(<Header />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders Header text', () => {
    render(<Header />);
    const linkToTest = screen.getByRole("link", { name: /Weather Anywhere/i });    
    expect(linkToTest.getAttribute("href")).toMatchInlineSnapshot(`"https://mmoresun.github.io/weather"`);
});


