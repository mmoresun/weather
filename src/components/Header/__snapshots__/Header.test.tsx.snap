// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly (snapshot) 1`] = `
<div
  className="MuiBox-root css-1qk4bij"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-py75jh-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1xouml8-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        <a
          href="https://mmoresun.github.io/weather"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          â›… Weather Anywhere
        </a>
      </div>
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit css-u44ztt-MuiButtonBase-root-MuiButton-root"
        href="https://github.com/mmoresun/weather/tree/master"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </header>
</div>
`;
